code/datums/components/anti_magic.dm

```diff
@@ -1,14 +1,18 @@
 /datum/component/anti_magic
 	var/magic = FALSE
 	var/holy = FALSE
-	var/tinfoil = FALSE
-	var/checkslot
+	var/psychic = FALSE
+	var/list/allowed_slots = list(SLOT_BACK, SLOT_WEAR_MASK, SLOT_HANDCUFFED, SLOT_HANDS, \
```

Comment:
We have item slot bitflags for this kind of thing

Reply:
I hope you aren't referring to item inventory slot bitflags such as ITEM_SLOT_OCLOTHING and ITEM_SLOT_GLOVES. From experience, mixing up inventory slot bitflags with equip slot defines can result in fuck ups, like guns being equippable on eyes to make an example.
I'm pretty sure you know better than this, so what do you suppose has to be done here?

Reply:
What I'm suggesting is that you just use the bitflags to store which slots are allowable, a list for this kind of thing is overkill. You can convert from the text define to bitflag easily enough.

So yes I am referring to the item slot defines. There's nothing wrong with them, especially here where you have full control over how they're used inside a component.

Reply:
Umh, quite right actually, I skipped over the existence of the slotdefine2slotbit() for a moment.